src/mapping_navigation/scripts/pure_pursuit.py

```diff
@@ -107,8 +96,8 @@ def search_target_index(self, state: State) -> Tuple[int, float]:
         if self.old_nearest_point_index is None:
             # search nearest point index
             # Mapping though all points in the path, and finding the difference in height and width to that point.
-            dx = [state.rear_x - icx for icx in self.cx]
-            dy = [state.rear_y - icy for icy in self.cy]
+            dx = [state.rear_x - point.coordinate[0] for point in self.path]
+            dy = [state.rear_y - point.coordinate[1] for point in self.path]
             #Getting the minimum hypotenues, aka the nearest point
             #For this firt time set it to the closest point.
             d = np.hypot(dx, dy)
```

Comment:
In the below two lines:
1. What does "ind = np.argmin(d)" : Isn't d a number? So what does argmin do?
2. self.old_nearest_point_index = np.argmin(d) is slightly cleaner, if that's your style

Reply:
d is an array of all the Euclidian distances, argmin is return the index to the element with the shortest distance, will try and update the naming to make this clearer